client/components/autocomplete_list/index.coffee

```diff
@@ -62,18 +61,20 @@ module.exports.AutocompleteList = AutocompleteList = React.createClass
     @$input.on 'typeahead:selected', @onSelect
 
   onSelect: (e, item) ->
-    @setState value: '', items: @state.items.concat [item]
+    @setState items: @state.items.concat [item]
     @props.selected? e, item, @state.items
+    @$input.val('')
+
+  onBlur: ->
+    @$input.val('')
```

Comment:
For some reason the input kept getting reset to the most recently selected/deleted item on blur, this seemed to kill it.